File: Wikipedia/Code/WMFArticleLanguagesSectionFooter.m
#import "WMFArticleLanguagesSectionFooter.h"
#import "Wikipedia-Swift.h"

@interface WMFArticleLanguagesSectionFooter ()

@property (strong, nonatomic) IBOutlet UILabel *titleLabel;
@property (strong, nonatomic) IBOutlet UIButton *addButton;
@property (strong, nonatomic) IBOutlet NSLayoutConstraint *topConstraint;

@end

@implementation WMFArticleLanguagesSectionFooter

- (void)setTitle:(NSString *)title {
    self.titleLabel.text = title;
}

- (void)awakeFromNib {
    [super awakeFromNib];
    UIView *backgroundView = [[UIView alloc] initWithFrame:self.bounds];
    self.backgroundView = backgroundView;
    [self.addButton setTitle:WMFLocalizedStringWithDefaultValue(@"welcome-languages-add-button", nil, nil, @"Add another language", @"Title for button for adding another language")
                    forState:UIControlStateNormal];
    [self wmf_configureSubviewsForDynamicType];
    [self applyTheme:[WMFTheme standard]];
}

- (void)setButtonHidden:(BOOL)hidden {
    [self.addButton setHidden:hidden];
    self.topConstraint.constant = hidden ? 10 : 0;
}

- (void)applyTheme:(WMFTheme *)theme {
    self.backgroundView.backgroundColor = theme.colors.baseBackground;
    self.titleLabel.textColor = theme.colors.secondaryText;
    [self.addButton setTitleColor:theme.colors.link forState:UIControlStateNormal];
}

- (IBAction)addLanguageButtonTapped:(id)sender {
    [self.delegate addLanguageButtonTapped];
}

@end

File: Wikipedia/Code/WMFArticleLanguagesSectionFooter.swift
class WMFArticleLanguagesSectionFooter : UITableViewHeaderFooterView, Themeable {
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var addButton: UIButton!
    override func awakeFromNib() {
        super.awakeFromNib()
        backgroundView = UIView.init(frame: bounds)
        addButton.setTitle(WMFLocalizedString("welcome-languages-add-button", value:"Add another language", comment:"Title for button for adding another language"), for: .normal)
        titleLabel.text = WMFLocalizedString("settings-primary-language-details", value:"The first language in this list is used as the primary language for the app. Changing this language will change daily content (such as Featured Article) shown on Explore.", comment:"Explanation of how the first preferred language is used. \"Explore\" is {{msg-wm|Wikipedia-ios-home-title}}.")
        wmf_configureSubviewsForDynamicType()
        apply(theme: .standard)
    }
    func apply(theme: Theme) {
        backgroundView?.backgroundColor = theme.colors.baseBackground
        titleLabel.textColor = theme.colors.secondaryText
        addButton.setTitleColor(theme.colors.link, for: .normal)
    }
}
